File: day2/Python/part2.py
import re
with open("input.txt", "r") as file:
    contents = file.read()
line_array = re.split("\n",contents) 
horizontal = 0
depth = 0
aim = 0
for line in line_array:
    array = re.split(" ",line)
    direction = array[0]
    num = int(array[1])
    if direction == "forward":
        horizontal += num
        depth += aim * num
    elif direction == "down":
        aim += num
    elif direction == "up":
        aim -= num
print(f"Total is {horizontal * depth}")    

File: day2/JS/part2.js
const lineReader = require("line-reader");

let horizontal = 0;
let depth = 0;
let aim = 0;

lineReader.eachLine("input.txt", (line, last) => {
  parseLineUpdated(line);
  if (last == true) console.log(horizontal * depth);
});

function parseLineUpdated(text) {
  let arr = text.split(" ");
  arr[1] = parseInt(arr[1]);
  switch (arr[0]) {
    case "forward":
      depth += (aim * arr[1]);
      horizontal += arr[1];
      break;
    case "down":
      aim += arr[1];
      break;
    case "up":
      aim -= arr[1];
      break;
  }
}
